modules/popup-stack/lib/PopupStack.ts

```diff
@@ -361,3 +466,24 @@ export function resetStack() {
 export const createAdapter = (adapter: Partial<typeof PopupStack>) => {
   stack._adapter = adapter;
 };
+
+// keep track of the element ourselves to avoid accidentally popping off someone else's stack
+// context
+let element: HTMLElement | null = null;
+
+// Where should this go? Each version of `PopupStack` on a page will add a listener. The
+// `PopupStack` should guard against multiple handlers like this simultaneously and there is no
+// lifecycle here.
+if (screenfull.isEnabled) {
+  screenfull.on('change', () => {
+    if (screenfull.isFullscreen) {
+      if (screenfull.element) {
+        element = screenfull.element as HTMLElement;
+        PopupStack.pushStackContext(element);
+      }
+    } else if (element) {
+      PopupStack.popStackContext(element);
+    }
+    console.log('change', screenfull.isFullscreen, stacks.length);
+  });
```

Comment:
Something would still need to call this method. Setting it to `null`  would mean the caller would have to do a nullable check. A global state variable would be the only way to ensure this method does not get called more than once, but it does put a weird cross-version dependency.

The way it is done here doesn't require a cross-version dependency, but does require `pushStackContext` to ignore requests from the same element. This requirement isn't only for this call though as pushing a new stack context could be from any number of components that could be placed on the same element. I think multiple calls to push a new stack context on the same element should be silently ignored (the side-effect is the same whether this was the first call or the 10th) rather than an error or other attempts to make sure the application only pushes a stack context element only once.

The question about the location of this call is around where could it be called where lifecycle is controlled. There are a few problems:
- How do we handle only adding a full screen listener if we have popups?
- How do we remove a full screen listener if there are no more popups?

I thought about other places this could go:
- In `Popper.tsx` in a `useEffect` hook
  - The problem here is the listener is added many more times than once per `PopupStack` import instance
- In `Popper.tsx` outside the component (side-effect)
  - The problem here is tree shaking. Tree shaking will theoretically remove non-reachable code. In practice, this probably isn't a problem because abuse of tree-shaking is so common. Many tree-shaking bundlers also need a special "pure" comment to flag removal and a side effect list this would not be flagged as "pure" and therefore not removed. There are higher optimizer settings that could still remove this code as "dead code" since there is no caller. For this reason, `@workday/canvas-kit-popup-stack` has been changed to a side-effect dependency to help remove chances of side-effect removal
- Only a method that is truly side-effect free
  - This would be a breaking change as the `PopupStack` library already relies on side effects to create a globally shared stack. It would mean the application would need to opt in to full screen support and could not be released as a patch. The reason the `PopupStack` relies on side-effects to remove the coordination required by a global call. It isn't obvious in larger application who is responsible for making this call. In a single app like Workday, that can be a platform team, but what about independent application development where an app may or may not be running in a context of the Workday application? That complexity would be pushed onto many developers. We made a decision to hide this complexity and simply using any popups would mean the system worked and it was designed with the multi-app use-case in mind.

I haven't thought of a better place to add the listener than `PopupStack.ts`. The lifecycle is tied to the page. This can mean slow memory leaks as any additional dynamic imports could add listeners that never get cleaned up, but that would be a very slow leak. The only real change that could mitigate slow memory leaks would be an additional global variable that checks if the listener is added, but that runs into the same multi-version dependency issue where the listener could never be changed by later versions. I still think this is the best place for this listener to be added to add support for full-screen without pushing the complexity to add development teams.